### PR TITLE
Add the calendar service boilerplate

### LINKED ISSUE #201: Add the Calendar service boilerplate
## Description

Add the calendar service boilerplate. This service will be used to fetch the existing calendars and calendar events using EventKit. For now just add the service boilerplate and define the `Calendar` and `CalendarEvent` structs, actual integration with EventKit will happen later.

The calendar service protocol should contain the following methods

```
func getAvailableCalendars() -> Single<[Calendar]>
func getCalendarEvents(between startDate: Date, and: endDate: Date, in calendars: [Calendar]) -> SingleCalendarEvent]>
```

Calendar properties:
- id: String
- name: String
- sourceName: String

CalendarEvent properties:
- id: String
- startTime: Date
- duration: TimeInterval
- description: String
- color: UIColor (or String?)
- calendarId: String

### PR DESCRIPTION
**Pull request by juanlaube**
## Info
This adds the calendar service boilerplate.

### Summary
This adds the calendar service boilerplate, for now it's just the structs for the data we need, a protocol, and a default implementation that doesn't do anything. We will replace that implementation later with the actual integration with EventKit.

### Technical considerations
Nothing remarkable to add here.

### Relationships
Closes #201

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] The code used custom templates

### Pointers
I don't have anything to add here, it's an empty service so far.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑

### CODE REVIEW

**Comment by heytherewill** [comment]
Do we really need an entire framework just for that? 🤔

**Comment by juanlaube** [comment]
Perhaps we could have a framework to hide all the iOS integrations later? For now I'd go with this and we can move it later.

File: Modules/CalendarService/CalendarService/CalendarService.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  CalendarService.h
+//  CalendarService
+//
+//  Created by Juan Laube on 5/28/20.
```
Comment (dshr) [inline Modules/CalendarService/CalendarService/CalendarService.h:5]: 🔥

File: Modules/CalendarService/CalendarServiceTests/CalendarServiceTests.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+//
+//  CalendarServiceTests.swift
+//  CalendarServiceTests
+//
+//  Created by Juan Laube on 5/28/20.
```
Comment (dshr) [inline Modules/CalendarService/CalendarServiceTests/CalendarServiceTests.swift:5]: 🔥

**Pull request merged**